### ISSUE TITLE
[Bug]: CrashLoopBackOff DataNode During BulkImport

### ISSUE THREAD

**Issue by akmalmasud96** [milvus-io/milvus#38895]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:v2.5.1
- Deployment mode(standalone or cluster): Aws EKS Cluster
- MQ type(rocksmq, pulsar or kafka):    pulsar
- SDK version(e.g. pymilvus v2.0.0rc2): 2.5.2
```

### Current Behavior

I have been importing the data into the milvus with parquet file, using bulk-import. However while importing, the data node is getting into the CrashLoopBackOff, and from the get_import_progress is showing the status for the files to be "Importing" from the last 5 hours. 
also Is it not that if the task is not imported the duration after 15 minutes import task will be expired ( default )?

### Steps To Reproduce

```markdown
resp = bulk_import(url="http://@localhost:19530",
                   collection_name = "collectionName",
                   partition_name = "partitionName",
                   files= [[path.parquet]] )
```
```


### Milvus Log

Attached are the logs of DataNode, which is in the CrashLoop.
[output.log](https://github.com/user-attachments/files/18281680/output.log)


### Anything else?

_No response_

**Comment by bigsheeper** [comment]
![image](https://github.com/user-attachments/assets/3509b5f1-1de6-4cc0-83eb-d0906496cd03)

**Comment by akmalmasud96** [comment]
The format of parquet file which i am importing is as follows,  
key: str, embeddings: float16

and when we create the parquet using pymilvus LocalBulkWriter, the type of embeddings is uint8 for float16 vector, does milvus also expect uint8 type for parquet ?

**Comment by xiaofan-luan** [comment]
> 我要导入的 parquet 文件的格式如下， key：str，embeddings：float16
> 
> 当我们使用 pymilvus LocalBulkWriter 创建 parquet 时，嵌入的类型为 float16 向量的 uint8，milvus 是否也期望 parquet 为 uint8 类型？

I think this is most likely a type mismatch.
How did you store fp16 in parquet?

**Comment by akmalmasud96** [comment]
@xiaofan-luan We have stored the FP16 NumPy array in Parquet, and the schema of the Parquet file, generated using PyArrow, is attached.

```
key: string
emb: list<element: halffloat>
  child 0, element: halffloat
```

**Comment by xiaofan-luan** [comment]
is there a sample file we can reproduce this issue?

**Comment by akmalmasud96** [comment]
Hi @xiaofan-luan , attach is the sample file. 

[000299438c7211efa73683014516d375.parquet.zip](https://github.com/user-attachments/files/18379943/000299438c7211efa73683014516d375.parquet.zip)

**Comment by xiaofan-luan** [comment]
@akmalmasud96 

For datatype float vector int milvus, we expect your vector to be a float vector.
For datatype fp16/bfp16, we expect it to be binary data.

From your data I parsed, it seems to be follow fp16. How did you define your milvus schema?
Does it claim to be BF16 vector as well?

actaully we would recommend to use milvus bulk writer to make sure you have the right data.

The reason fp16 and fp32 is different is because parquet by native don't support fp16/bf16 format. 

Check importutilv2 -> parquet _> field_reader.go for details

**Comment by akmalmasud96** [comment]
@xiaofan-luan 
Yes, the data we have is in fp16 format, and the milvus-schema is also defined to be fp16.
Currently the routine makes the data in the shared sample format, and when we would have to bulk_insert, we just move it to the milvus-s3-bucket. If we have to use bulk_writer then it would increase the parquet file preparation time, as we have to insert millions of data each day. 
We will explore the bulk-writer that how it writes the data in parquet, we will start preparing the data in the same format. 
Secondly, the milvus data node is going to the crash-loop instead of giving the error. I think this error is not handled if the format of file is not correct, rather than going to crash-loop. 

Also, the task should get expired after 15 minutes, however, it doesnt get expired even after hours.

**Comment by xiaofan-luan** [comment]
> [@xiaofan-luan](https://github.com/xiaofan-luan) Yes, the data we have is in fp16 format, and the milvus-schema is also defined to be fp16. Currently the routine makes the data in the shared sample format, and when we would have to bulk_insert, we just move it to the milvus-s3-bucket. If we have to use bulk_writer then it would increase the parquet file preparation time, as we have to insert millions of data each day. We will explore the bulk-writer that how it writes the data in parquet, we will start preparing the data in the same format. Secondly, the milvus data node is going to the crash-loop instead of giving the error. I think this error is not handled if the format of file is not correct, rather than going to crash-loop.
> 
> Also, the task should get expired after 15 minutes, however, it doesnt get expired even after hours.

1. i checked the parquet file and it's binary files. I don't check what's the reason it's not correct but unfortunately it's not same as what we expected. You will need to check milvus writers logic on how to import the right file format. FP16 is very special because it is not natively defined in parquet.
2. since the file is wrong, and format itself is binary, the parquet reader panic and it's hard for milvus to handle this logic.
3. you can cancel the bulkinsert job manually. It is hang by design and wait for executor to deal with it for quite a long time

**Comment by akmalmasud96** [comment]
@xiaofan-luan We have implemented it with the bulk-writer and its working now. Thanks
